File: Baekjoon Online Judge/1 ~ 9999/9095/boj-9095.cpp
/*
    Copyright(c) 2019. Minho Cheon. All rights reserved.
*/

#include <iostream>
#include <cstring>

using namespace std;

int cache[20];

int func(int num) {
	if (num == 0) {
		return 1;
	}

	int& ret = cache[num];
	if (ret != -1) {
		return ret;
	}

	ret = 0;
	for (int i = 1 ; i <= 3 ; ++i) {
		if (num >= i) {
			ret += func(num - i);
		}
	}

	return ret;
}

int main() {
	int t, n;
	cin >> t;

	memset(cache, -1, sizeof(cache));
	while (t--) {
		cin >> n;
		cout << func(n) << endl;
	}
}

File: Baekjoon Online Judge/1 ~ 9999/9095/boj-9095.py
def func(n):
    if n == 0:
        return 1
    ret = 0
    for i in range(1, 4):
        if n - i >= 0:
            ret += func(n - i)
    return ret

t = int(input())
for _ in range(t):
    n = int(input())
    print(func(n))
